integration/spark/spark3/src/main/java/io/openlineage/spark3/agent/lifecycle/plan/column/ColumnLevelLineageUtils.java

```diff
@@ -55,7 +55,21 @@ public static Optional<OpenLineage.ColumnLineageDatasetFacet> buildColumnLineage
     OpenLineage.ColumnLineageDatasetFacetBuilder facetBuilder =
         olContext.getOpenLineage().newColumnLineageDatasetFacetBuilder();
 
-    facetBuilder.fields(context.getBuilder().build());
+    boolean datasetLineageEnabled =
+        context
+            .getOlContext()
+            .getOpenLineageConfig()
+            .getColumnLineageConfig()
+            .isDatasetLineageEnabled();
+    if (!datasetLineageEnabled) {
+      log.warn(
+          "DEPRECATION WARNING: The columnLineage.datasetLineageEnabled configuration is set to false. This flag will default to false in the future versions and soon will be removed (defaulting to true).");
```

Comment:
good deprecation message, should it be `warn` though? It's just information so maybe `info`, `WARNING` in the logs should suggest that something could fail here

Reply:
It is going to fail in just a couple of releases.

Reply:
so in couple of releases it will be a good warning :)

Reply:
Now it warns that the default is going to change and your events will be affected

Reply:
I mean, it's not that big of a deal since it will be removed relatively soon. 

Nonetheless warning should tell you about something that may cause failure, so I wouldn't log warnings about something concerning future versions (which current users may not upgrade to anytime soon)